### DISCUSSION TITLE
Why are the results between these two simulations different?

### DISCUSSION THREAD

**Discussion by SengerM** [thliebig/openEMS-Project#238] [Q&A Usage] [answered]
I am taking my first steps with openEMS, and this kind of simulations in general. I am interested in simulating arbitrary geometries, created with other software (FreeCAD). I decided to start simple with [the simple patch antenna tutorial](https://docs.openems.de/python/openEMS/Tutorials/Simple_Patch_Antenna.html), which runs perfectly fine in my PC. Then, I decided to replace the substrate with an STL file generated with FreeCAD. I draw a parallelepiped with the same dimensions as the substrate and exported it into STL. I changed the lines where the substrate is defined:
```python
substrate = CSX.AddMaterial('substrate', epsilon=substrate_epsR, kappa=substrate_kappa)
whatever = substrate.AddPolyhedronReader('substrate.stl', priority=0)
whatever.ReadFile()
```
and run the simulation. To my surprise, the results are quite different (the impedance value is not anymore 50 Ω):
![image](https://github.com/user-attachments/assets/cc056bfd-415b-4eec-8823-2a24e9762ad9)

What is the reason behind this?

All my files: [simple_patch_antenna.zip](https://github.com/user-attachments/files/17391367/simple_patch_antenna.zip)

**Comment by thliebig** [comment]
Have you loaded and looked at your structures in AppCSXCAD (the graphical display tool) to make sure the stl has the same position and dimension as you box previously?  usually stl is defined in mm and you may need to scale it to fit to the structure?

**Reply by SengerM** [reply]
I still do not manage to make AppCSXCAD work, so I am working blind. I will try to install it again, as it seems to be very helpful for this.

**Comment by LubomirJagos42** [comment, marked as answer]
looking into AppCSXCAD this is your model
![image](https://github.com/user-attachments/assets/f5b6f82e-3ccc-49b6-ab79-849fa7e58031)
you have gnd plane same size as patch due wrong dimensions for it, correct this and you got same results
```python
# create ground (same size as substrate)
gnd = CSX.AddMetal( 'gnd' ) # create a perfect electric conductor (PEC)
#start[2]=0    #original just change Z to 0mm
#stop[2] =0    #original just change Z to 0mm
start = [-30, -30, 0]    #LuboJ modified, gnd plane must be bigger than patch, looked into FreeCAD what is substrate width
stop = [30, 30, 0]    #LuboJ modified, gnd plane must be bigger than patch, looked into FreeCAD what is substrate height
gnd.AddBox(start, stop, priority=10)   
```

**Reply by SengerM** [reply]
Thanks a lot for checking, I see the issue, I accidentally removed the re-definition of `start` and `stop` which was reused for the ground plane.
